java/vector/src/test/java/org/apache/arrow/vector/TestDecimalVector.java

```diff
@@ -161,6 +161,34 @@ public void testWriteBigEndian() {
     }
   }
 
+  @Test
+  public void testLongReadWrite() {
+    try (DecimalVector decimalVector = TestUtils.newVector(DecimalVector.class, "decimal",
+            new ArrowType.Decimal(38, 0), allocator)) {
+      decimalVector.allocateNew();
+      long long1 = 0L;
```

Comment:
can be simplified by storing the values in an array and iterating over it to do the set and assert..